pkg/agent/svid/rotator_test.go

```diff
@@ -152,14 +157,93 @@ func TestRotator(t *testing.T) {
 	}
 }
 
+func TestRotationFails(t *testing.T) {
+	caCert, caKey := testca.CreateCACertificate(t, nil, nil)
+
+	expiredStatus := status.New(codes.PermissionDenied, "agent is expired")
```

Comment:
nit:
Even if it's a test, it may be confusing
```suggestion
	expiredStatus := status.New(codes.PermissionDenied, "agent is not active")
```